src/librustc_trans/trans/mir/constant.rs

```diff
@@ -49,8 +68,11 @@ impl<'bcx, 'tcx> MirContext<'bcx, 'tcx> {
     {
         let constant_ty = bcx.monomorphize(&constant.ty);
         match constant.literal {
-            mir::Literal::Item { .. } => {
-                unimplemented!()
+            mir::Literal::Item { def_id, substs } => {
```

Comment:
Yea sure, ended up seeing it later and it's definitely more complete. I'll rebase on top of that instead.